Enterprise-CMCS/macpro-mdct-mcr
# [MDCT-2419] - Modify the date optional schema to actually be optional

### Description
The `dateOptional` schema on the frontend should match the `date` schema, with the `required` test remove and `nullable` and `notRequired` added.

### Related ticket(s)


MDCT-2419

---
### How to test
1. Create a new MLR submission
2. Add a new report entity
3. Select "Yes" for question 4.1
4. Select "No" for question 4.7
5. Verify nested dates do not trigger validation.

### Important updates
Changes the `dateOptional` schema, but this is rarely used anyways.


---
### Author checklist


---


_convert to a different template: [test → val](?expand=1&template=test-to-val-deployment.md)_ | _[val → prod](?expand=1&template=val-to-prod-deployment.md)_
